tests/hpc/rasdaemon.pm

```diff
@@ -21,27 +29,65 @@ use warnings;
 use testapi;
 use utils;
 
+sub inject_error {
+    # load kernel module
+    assert_script_run('modprobe mce-inject');
+    # Inject some software errors
+    script_run('echo 0x9c00410000080f2b > /sys/kernel/debug/mce-inject/status');
+    script_run('echo d5a099a9 > /sys/kernel/debug/mce-inject/addr');
+    script_run('echo 4 > /sys/kernel/debug/mce-inject/bank');
+    script_run('echo 0xdead57ac1ba0babe > /sys/kernel/debug/mce-inject/misc');
+    script_run("echo \"sw\" > /sys/kernel/debug/mce-inject/flags");
+}
+
 sub run {
     my $self = shift;
 
     zypper_call('in rasdaemon');
 
     assert_script_run('! ras-mc-ctl --status');
+
     systemctl('start rasdaemon');
+    systemctl('is-active rasdaemon');
 
     # Validating output of 'ras-mc-ctl --mainboard'
     my $mainboard_output = script_output('ras-mc-ctl --mainboard');
-    die 'Not expected mainboard - ' . $mainboard_output unless $mainboard_output =~ /mainboard/;
+    record_info('INFO', $mainboard_output);
+    die('Not expected mainboard - ' . $mainboard_output)
+      unless ($mainboard_output =~ /mainboard/);
 
-    # Validating output of 'ras-mc-ctl --summary'
+    # Validating output of 'ras-mc-ctl --summary' with assumption that no errors exists
     my $summary_output = script_output('ras-mc-ctl --summary');
-    die 'Not expected summary - ' . $summary_output
-      unless $summary_output =~ /No Memory errors/ && $summary_output =~ /No PCIe AER errors/ && $summary_output =~ /No MCE errors/;
+    record_info('INFO', $summary_output);
 
-    # Validating output of 'ras-mc-ctl --errors'
+    die('Not expected summary - ' . $summary_output)
+      unless ($summary_output =~ /No Memory errors/
+      && $summary_output =~ /No PCIe AER errors/ && $summary_output =~ /No MCE errors/);
+
+    # Validating output of 'ras-mc-ctl --errors' with assumption that no errors exists
+    my $empty_error_output = script_output('ras-mc-ctl --errors');
+    record_info('INFO', $empty_error_output);
+
+    die('Not expected error - ' . $empty_error_output)
+      unless ($empty_error_output =~ /Memory errors/
+      && $empty_error_output =~ /PCIe AER errors/ && $empty_error_output =~ /No MCE errors/);
+
+    ##TODO: run only for Intel
+    # x86_64 check: Validating output of 'ras-mc-ctl --errors' after MCE error is injected
```

Comment:
so maybe you might wrap this part into something like 
`if(check_var('ARCH','x86_64'))`

Reply:
sure thing. It isn't ready PR ;-) but yeah there if be such if - most likely - and something for arm too.

Reply:
added this immediately and new revision pushed.